### PR TITLE
HS-379: Add support for config updates / event based discovery of devices

### PR DESCRIPTION
**Pull request by cgorantla**
## Description
Add support for sending events for config updates
Add event based discovery for Traps

## Jira link(s)
- https://issues.opennms.org/browse/HS-379
- https://issues.opennms.org/browse/HS-364

## Checklist
* [x] Follows Horizon Stream's [development guidelines.](https://github.com/OpenNMS/horizon-stream/wiki/Development-Guidelines)
* [x] Appropriate reviewer(s) have been selected.
* [x] Jira issue(s) have been updated to "In Review".
* [] Includes [appropriate tests.](https://github.com/OpenNMS/horizon-stream/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.

### CODE REVIEW

**Comment by cgorantla** [comment]
To validate this:

Update config by setting  `newSuspectOnTrap` to true on Core from Rest call : http://localhost:18181/config/snmp-traps

And send a trap, that trap should generate event on Core and also generate another new suspect event.
And that event should add a device as discovered device and start collecting metrics for that new device.

File: platform/config/service/src/main/java/org/opennms/horizon/config/service/impl/DefaultConfigService.java [hunk 1]
```diff
@@ -44,34 +49,48 @@
...
     }
 
 
     @Override
-    public void addConfig(String configName, String jsonConfig) {
+    public void addConfig(String configName, String jsonConfig, String source) {
         addConfigNameToJsonStore(configName);
         try {
+            Optional<String> existingConfig = getConfig(configName);
+            if (existingConfig.isPresent() && existingConfig.get().equals(jsonConfig)) {
```
Comment (yangli2014) [inline platform/config/service/src/main/java/org/opennms/horizon/config/service/impl/DefaultConfigService.java:68]: We added the configure name to the JSon store at line 65 first then we check if the config exists, it seems will always return true here.

File: platform/config/service/src/main/java/org/opennms/horizon/config/service/impl/DefaultConfigService.java [hunk 2]
```diff
@@ -44,34 +49,48 @@
...
+            LOG.error("Exception while parsing config for {}", configName, e);
             throw new IllegalArgumentException("Invalid json " + jsonConfig);
         }
         jsonStore.put(configName, jsonConfig, CONFIG);
+        sendConfigUpdatedEvent(configName, source);
     }
 
     @Override
-    public void updateConfig(String configName, String jsonConfig) {
+    public void updateConfig(String configName, String jsonConfig, String source) {
```
Comment (yangli2014) [inline platform/config/service/src/main/java/org/opennms/horizon/config/service/impl/DefaultConfigService.java:81]: This method has exactly the same code as ```addConfig``` above.

**Review by yangli2014 — APPROVED** [review]
approved with minor comments

File: platform/config/service/src/main/java/org/opennms/horizon/config/service/impl/DefaultConfigService.java (hunk 1 above)
Comment (cgorantla) [reply platform/config/service/src/main/java/org/opennms/horizon/config/service/impl/DefaultConfigService.java:68]: Config name is in separate context in json store and Config may not be stored if it is not valid Json.
Config names are stored separately not along with config.

**Pull request merged**